docs/book/tutorials/kubernetes-on-vsphere-with-kubeadm.md

```diff
@@ -62,88 +141,61 @@ sudo tee /etc/docker/daemon.json >/dev/null <<EOF
-
-```sh
-sudo sysctl net.bridge.bridge-nf-call-iptables=1
+# mkdir -p /etc/systemd/system/docker.service.d
 ```
 
-## Enabling the VMware vSphere Cloud Provider
-
-### On the master(s)
-
-#### Create your `vsphere.conf` file with vCenter details
-
-For reference, the vCenter configuration looks like below (you can correlate the values in the UI to the values in the `vsphere.conf` file below):
-
-![vCenter](/docs/images/vCenter.png)
-
-Edit the below command to fill in your vCenter details before running.
-
-If you don't have a folder created with your kubernetes node VMs added we can do that quickly with `govc` (note, change `vSAN-DC` to your Datacenter name in vCenter):
-
-*N.B: The below command assumes your K8s VMs are named in the pattern `k8s-*`, if they have a different naming schema, please adjust the command below appropriately.*
-
-```sh
-govc folder.create /vSAN-DC/vm/k8s
-govc object.mv /vSAN-DC/vm/k8s-\* /vSAN-DC/vm/k8s
+And to complete, restart docker to pickup the new parameters.
 ```
+# systemctl daemon-reload
+# systemctl restart docker
+```
+Docker is now installed.
+### Install Kubelet, Kubectl, Kubeadm
+The next step is to install the main Kubernetes components on each of the nodes.
 
-Details on [syntax for the below vsphere.conf file can be found here](https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/existing.html). It is important to note, whatever `VM folder` you specify below needs to be pre-created in your vCenter, in my case the folder is called `k8s`.
-
-```sh
-sudo tee /etc/kubernetes/vsphere.conf >/dev/null <<EOF
-[Global]
-user = "administrator@vsphere.local"
-password = "Admin!23"
-port = "443"
-insecure-flag = "1"
-
-[VirtualCenter "10.198.17.154"]
-datacenters = "vSAN-DC"
+kubeadm is the command to bootstrap the cluster. It runs on the master and all worker nodes.
+kubelet is the component that runs on all nodes in the cluster and performs such tasks as starting pods and containers.
+kubectl is the command line utility to communicate with your cluster. It runs only on the master node.
+For Ubuntu distributions, a specific version can be installed with specifying version of the package name, e.g. ‘apt install -qy kubeadm=1.14.2-00 kubelet=1.14.2-00 kubectl=1.14.2-00’. These should be the minimum versions installed. 
 
-[Workspace]
-server = "10.198.17.154"
-datacenter = "vSAN-DC"
-default-datastore = "vsanDatastore"
-resourcepool-path = "vSAN-Cluster/Resources"
-folder = "k8s"
+First, the Kubernetes repository needs to be added to apt.
+```
+# curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
+```
+```
+# cat <<EOF >/etc/apt/sources.list.d/kubernetes.list
+deb https://apt.kubernetes.io/ kubernetes-xenial main
+EOF
+```
 
-[Disk]
-scsicontrollertype = pvscsi
+Next, install kubelet, kubectl and kubeadm for cluster spinup.
+```
+# apt update
```

Comment:
Avoid mixing `apt` and `apt-get` I'd say just stick with `apt` throughout.

Reply:
OK